scripts/brp-strip

```diff
@@ -7,11 +7,43 @@ fi
 STRIP=${1:-strip}
 NCPUS=${RPM_BUILD_NCPUS:-1}
 
+# 32 was chosen as a compromise between reducing the overhead of starting new
+# processes and distributing the work load evenly over as much processors as
+# possible
+MAX_ARGS=32
+
 case `uname -a` in
 Darwin*) exit 0 ;;
 *) ;;
 esac
 
 # Strip ELF binaries
-find "$RPM_BUILD_ROOT" -type f \! -regex "${RPM_BUILD_ROOT}/*usr/lib/debug.*" \! -name "*.go" -print0 | \
-    xargs -0 -r -P$NCPUS -n32 sh -c "file \"\$@\" | sed -n -e 's/^\(.*\):[ 	]*ELF.*, not stripped.*/\1/p' | grep -v 'no machine' | xargs -I\{\} $STRIP -g \{\}" ARG0
+# Create a tmp file for storing file paths
+# Below is the explanation of commands in the order of their appearance
+# Ignore /usr/lib/debug entries
+# Ignore all go(guile objects & golang) files
+# Run the file command to find relevant non-stripped binaries, with bundle size of 32
+# Ignore all 'no machine' files
+# Only operate on non-stripped binaries
+# Get inode & file names to help eliminate hard links (%i %n)
+tmp_fn="$(mktemp)"
+find "$RPM_BUILD_ROOT" -type f \
+  ! -regex "${RPM_BUILD_ROOT}/?usr/lib/debug/.*" \
+  ! -name "*.go" -print0 | \
+  xargs -0 -r -P${NCPUS} -n${MAX_ARGS} sh -c "file \"\$@\" | \
+  sed -n -e 's/^\(.*\):[[:space:]]*ELF.*, not stripped.*/\1/p' | \
+  grep -v 'no machine' | \
+  xargs -r stat -c '%i %n'" ARG0 > "${tmp_fn}"
```

Comment:
Isn't this overwriting `${tmp_fn}` from multiple processes? Using `>>` for the redirection should solve this. Still having this tested with a significant number of binaries would make be sleep better.

Reply:
Yeah, I got this thought. Surprisingly it doesn't and I tested this with a random find command with xargs like the way it's used here. Anyway, I will do this change.